### ISSUE TITLE
Build fail on 20.05 with: error: catching polymorphic type 'class armnn::Exception' by value

### ISSUE THREAD

**Issue by ggardet** [ARM-software/armnn#398]
Build fail with GCC10 on version 20.05 with: `error: catching polymorphic type 'class armnn::Exception' by value`
```
[  650s] /home/abuild/rpmbuild/BUILD/armnn-20.05/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp: In function 'int main(int, char**)':
[  650s] /home/abuild/rpmbuild/BUILD/armnn-20.05/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp:37:32: error: catching polymorphic type 'class armnn::Exception' by value [-Werror=catch-value=]
[  650s]    37 |         catch(armnn::Exception e)                                  \
[  650s]       |                                ^
[  650s] /home/abuild/rpmbuild/BUILD/armnn-20.05/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp:212:5: note: in expansion of macro 'CHECK_OK'
[  650s]   212 |     CHECK_OK(LoadModel(backboneModelFile.c_str(), *parser, *runtime, backboneId, {"GpuAcc", "CpuRef"}));
[  650s]       |     ^~~~~~~~
[  650s] /home/abuild/rpmbuild/BUILD/armnn-20.05/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp:37:32: error: catching polymorphic type 'class armnn::Exception' by value [-Werror=catch-value=]
[  650s]    37 |         catch(armnn::Exception e) 
```

**Comment by ggardet** [comment]
Cc @DerekLambertiOfArm @TeresaARM

**Comment by ggardet** [comment]
The following patch  fixes it:
```
--- armnn-20.05.orig/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp	2020-06-08 08:53:31.543112449 +0200
+++ armnn-20.05/tests/TfLiteYoloV3Big-Armnn/TfLiteYoloV3Big-Armnn.cpp	2020-06-08 08:54:20.503535598 +0200
@@ -34,7 +34,7 @@ static const int GENERAL_ERROR = -100;
             auto r_local = v;                       \
             if (r_local != 0) { return r_local;}    \
         }                                           \
-        catch(armnn::Exception e)                                  \
+        catch(armnn::Exception &e)                                  \
         { \
             ARMNN_LOG(error) << "Oops: " << e.what(); \
             return GENERAL_ERROR; \
```

**Comment by MatthewARM** [comment]
Fixed in 855a47b1b0a
